internal/providers/applehv/applehv.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2023 Red Hat
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package applehv
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os/exec"
+
+	"github.com/coreos/ignition/v2/config/v3_5_experimental/types"
+	"github.com/coreos/ignition/v2/internal/distro"
+	"github.com/coreos/ignition/v2/internal/platform"
+	"github.com/coreos/ignition/v2/internal/providers/util"
+	"github.com/coreos/ignition/v2/internal/resource"
+	"github.com/coreos/vcontext/report"
+	"github.com/mdlayher/vsock"
+)
+
+func init() {
+	platform.Register(platform.Provider{
+		Name:  "applehv",
+		Fetch: fetchConfig,
+	})
+}
+
+func fetchConfig(f *resource.Fetcher) (types.Config, report.Report, error) {
+	_, err := f.Logger.LogCmd(exec.Command(distro.ModprobeCmd(), "vsock"), "Loading vsock kernel module")
+	if err != nil {
+		f.Logger.Err("Failed to install vsocker kernel module: %v", err)
+		errors := fmt.Errorf("Failed to install vsock kernel module: %v", err)
+		return types.Config{}, report.Report{}, errors
+	}
+	conn, err := vsock.Dial(2, 1024, &vsock.Config{})
```

Comment:
brief googling has not indicated why port 1024 is used? is it specific or just picked by us?

Reply:
pick by me; it can be changed if there is a precedent.  i would also have to change some docs and podman code but that is fair game

Reply:
Let's document that 1024 is the port that the HTTP server on the other end will have to listen to (from https://pkg.go.dev/github.com/mdlayher/vsock#Dial). Is this something set by the applehv platform or decided by podman?

Reply:
something we decided, will document